sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -1639,13 +1680,13 @@ export class Ground_Block implements GroundBlock {
     return {
       name: this.name,
       time:
-        this._absoluteTime !== null
-          ? { type: TimingTypes.ABSOLUTE, tag: instantToDoy(this._absoluteTime) }
-          : this._epochTime !== null
-          ? { type: TimingTypes.EPOCH_RELATIVE, tag: durationToHms(this._epochTime) }
-          : this._relativeTime !== null
-          ? { type: TimingTypes.COMMAND_RELATIVE, tag: durationToHms(this._relativeTime) }
-          : { type: TimingTypes.COMMAND_COMPLETE },
+          this._absoluteTime !== null
+              ? { type: TimingTypes.ABSOLUTE, tag: instantToDoy(this._absoluteTime) }
+              : this._epochTime !== null
+                  ? { type: TimingTypes.EPOCH_RELATIVE, tag: durationToHms(this._epochTime) }
+                  : this._relativeTime !== null
+                      ? { type: TimingTypes.COMMAND_RELATIVE, tag: durationToHms(this._relativeTime) }
+                      : { type: TimingTypes.COMMAND_COMPLETE },
```

Comment:
These nested ternary checks are hard to read, same as my other comment if there's a way to make it more readable!

Reply:
Yeah, the nested turnary's need to turn off the default formatting. They're reaonably readable if formatted as a switch-expression

ex.
```
r.steps[0].type === StepType.Command ? CommandStem.fromSeqJson(r.steps[0] as CommandStem,localNames, parameterNames)
// @ts-ignore : 'GroundBlock' found in JSON Spec
: r.steps[0].type === StepType.GroundBlock ? Ground_Block.fromSeqJson(r.steps[0] as GroundBlock)
// @ts-ignore : 'GroundEvent' found in JSON Spec
: r.steps[0].type === StepType.GroundEvent ? Ground_Event.fromSeqJson(r.steps[0] as GroundEvent)
// @ts-ignore : 'GroundBlock' found in JSON Spec
: r.steps[0].type === StepType.Activate ? ActivateStep.fromSeqJson(r.steps[0] as ActivateStep)
// @ts-ignore : 'GroundBlock' found in JSON Spec
: r.steps[0].type === StepType.Load ? LoadStep.fromSeqJson(r.steps[0] as LoadStep)
: r.steps[0],
```

or

```
r.steps[0].type === StepType.Command
  ? CommandStem.fromSeqJson(r.steps[0] as CommandStem,localNames, parameterNames)
: r.steps[0].type === StepType.GroundBlock
  // @ts-ignore : 'GroundBlock' found in JSON Spec
  ? Ground_Block.fromSeqJson(r.steps[0] as GroundBlock)
: r.steps[0].type === StepType.GroundEvent
  // @ts-ignore : 'GroundEvent' found in JSON Spec
  ? Ground_Event.fromSeqJson(r.steps[0] as GroundEvent)
: r.steps[0].type === StepType.Activate
  // @ts-ignore : 'ActivateStep' found in JSON Spec
  ? ActivateStep.fromSeqJson(r.steps[0] as ActivateStep)
: r.steps[0].type === StepType.Load
  // @ts-ignore : 'LoadStep' found in JSON Spec
  ? LoadStep.fromSeqJson(r.steps[0] as LoadStep)
: r.steps[0],
```

Reply:
https://github.com/tc39/proposal-pattern-matching very much needed

Reply:
even Java is getting pattern matching [soon](https://openjdk.org/jeps/441) 🥺

Reply:
I took this style and updated the code throughout the preface file to match:

```
r.steps[0].type === StepType.Command
  ? CommandStem.fromSeqJson(r.steps[0] as CommandStem,localNames, parameterNames)
: r.steps[0].type === StepType.GroundBlock
  // @ts-ignore : 'GroundBlock' found in JSON Spec
  ? Ground_Block.fromSeqJson(r.steps[0] as GroundBlock)
: r.steps[0].type === StepType.GroundEvent
  // @ts-ignore : 'GroundEvent' found in JSON Spec
  ? Ground_Event.fromSeqJson(r.steps[0] as GroundEvent)
: r.steps[0].type === StepType.Activate
  // @ts-ignore : 'ActivateStep' found in JSON Spec
  ? ActivateStep.fromSeqJson(r.steps[0] as ActivateStep)
: r.steps[0].type === StepType.Load
  // @ts-ignore : 'LoadStep' found in JSON Spec
  ? LoadStep.fromSeqJson(r.steps[0] as LoadStep)
: r.steps[0],
```